keps/sig-api-machinery/1965-kube-apiserver-identity/README.md

```diff
@@ -122,24 +122,22 @@ will only delay the storage migration for the same period of time.
 
 The [kubelet heartbeat](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/589-efficient-node-heartbeats)
 logic [already written](https://github.com/kubernetes/kubernetes/tree/master/pkg/kubelet/nodelease)
-will be re-used. The heartbeat controller will be added to kube-apiserver in a
-post-start hook.
-
-Each kube-apiserver will run a lease controller in a post-start-hook to refresh
-its Lease every 10s by default. A separate controller named [storageversiongc](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/storageversiongc/gc_controller.go)
-running in kube-controller-manager will watch the Lease API using an informer, and
-periodically resync its local cache. On processing an item, the `storageversiongc` controller
-will delete the Lease if the last `renewTime` was more than `leaseDurationSeconds` ago (default to 1h).
-The default `leaseDurationSeconds` is chosen to be way longer than the default
-refresh period, to tolerate clock skew and/or accidental refresh failure. The
-default resync period is 1h. By default, assuming negligible clock skew, a Lease
-will be deleted if the kube-apiserver fails to refresh its Lease for one to two
-hours. The `storageversiongc` controller will run in kube-controller-manager, to leverage leader
-election and reduce conflicts.
-
-The refresh rate, lease duration will be configurable through kube-apiserver
-flags. The resync period will be configurable through a kube-controller-manager
-flag.
+will be re-used. The lease creation and heart beat will be managed by the `start-kube-apiserver-identity-lease-controller`
+post-start-hook and expired leases will be garbage collected by the `start-kube-apiserver-identity-lease-garbage-collector`
+post-start-hook in kube-apiserver. The refresh rate, lease duration will be configurable through kube-apiserver
+flags
+
+The format of the lease ID will be `kube-apiserver-<UUID>`. The UUID is newly generated on every start-up. This ID format is preferred
```

Comment:
Did you persuade @deads2k about this? Or will there be a follow-up?

Reply:
I think there needs to be more follow-up, for now trying to update the KEP to reflect current state since it got very stale.

Reply:
I am still not sure what specific problem(s) having static API server IDs is supposed to solve.

Reply:
We should have this discussion maybe in a SIG meeting :)

Reply:
Good point -- I added it to this week's agenda

Reply:
if we do not use graceful release and we have a long enough O(hours), I might be able to live without static names.  I want to avoid the case where the overall view of kms write keys ignores an active kube-apiserver that is restarting.